planner/datasource.go

```diff
@@ -31,8 +31,13 @@ func (p *Planner) getCollectionScanPlan(mapperSelect *mapper.Select) (planSource
 		return planSource{}, err
 	}
 
+	var isQuerySource bool
+	if len(col.Description().Sources) != 0 {
+		_, isQuerySource = col.Description().Sources[0].(*client.QuerySource)
```

Comment:
todo: A comment here to why it is always the first Source (i.e. `0`) would be nice. If is documented elsewhere, suggest making this into a variable(`0`) and documenting it once there and using that wherever.

Reply:
It is `0` atm because len === 0 atm and we don't support anything else atm.  When that changes the `0` probably will.